python/TestHarness/TestHarness.py

```diff
@@ -484,6 +482,7 @@ def handleJobStatus(self, job):
             # Print results and perform any desired post job processing
             if job.isFinished():
                 tester = self.normalizeStatus(job)
+                self.error_code = self.error_code | job.getStatusCode() | tester.getStatusCode()
```

Comment:
Again, this should be moved into the Job. There's too much "leaky encapsulation" going on here. Don't make the TestHarness dig or figure this kind of stuff out. The Job should do it and then the TestHarness can just make a simple query to get the desired information back.